release/k3s/k3s.go

```diff
@@ -638,3 +640,44 @@ func cleanGitRepo(dir string) error {
 
 	return nil
 }
+
+func (r *Release) CreateRelease(ctx context.Context, client *github.Client, rc bool) error {
+	rcNum := 1
+	name := r.NewK8SClient + "+" + r.NewK3SVersion
+	oldName := r.OldK8SVersion + "+" + r.OldK8SVersion
+	for {
+		if rc {
+			name = r.NewK8SVersion + "-rc" + strconv.Itoa(rcNum) + "+" + r.NewK3SVersion
+		}
+		opts := &repository.CreateReleaseOpts{
+			Repo:         k3sRepo,
+			Name:         name,
+			Prerelease:   rc,
+			Branch:       r.ReleaseBranch,
+			Draft:        !rc,
+			ReleaseNotes: "",
+		}
+		if !rc {
+			latestRc, err := release.LatestRC(ctx, k3sRepo, r.NewK8SVersion, client)
+			buff, err := release.GenReleaseNotes(ctx, k3sRepo, latestRc, oldName, client)
+			if err != nil {
+				return err
+			}
+			opts.ReleaseNotes = buff.String()
+		}
+		_, err := repository.CreateRelease(ctx, client, opts)
+		if err != nil {
+			githubErr := err.(*github.ErrorResponse)
```

Comment:
Can we use errors.Is or errors.As there instead of the cast?

Reply:
I checked github library that we use, and unfortunately I cant find good documentation for the error code, it seemed to be unstructured, and I found that comment in the code:
```
GitHub error responses structure are often undocumented and inconsistent.
Sometimes error is just a simple string (Issue #540).
In such cases, Message represents an error message as a workaround.
```
here is the referenced issue https://github.com/google/go-github/issues/540